lib/src/supabase_auth.dart

```diff
@@ -300,12 +300,101 @@ extension GoTrueClientSignInProvider on GoTrueClient {
       scopes: scopes,
       queryParams: queryParams,
     );
-    final url = Uri.parse(res.url!);
-    final result = await launchUrl(
-      url,
-      mode: authScreenLaunchMode,
-      webOnlyWindowName: '_self',
+    final uri = Uri.parse(res.url!);
+
+    bool? result;
+    if (authScreenLaunchMode == LaunchMode.inAppWebView) {
```

Comment:
After making `context` optional, we might need to do this to avoid breaking change for users that are already using `inAppWebView`. 
```suggestion
    if (authScreenLaunchMode == LaunchMode.inAppWebView && context != null) {
```

Reply:
I think we should put an assertion error inside the if statement because it could be hard or non intuitive for developers to understand that passing an optional `BuildContext` can cause a different behavior.

Maybe the breaking change is mandatory?

Reply:
I agree that it would feel unintuitive, but we still cannot do a breaking change without a major version bump, which we don't plan to do just not quite yet. We can update the default behavior of the method to a more intuitive one in the next major launch.

Reply:
> I agree that it would feel unintuitive, but we still cannot do a breaking change without a major version bump, which we don't plan to do just not quite yet. We can update the default behavior of the method to a more intuitive one in the next major launch.

Ok, I’ll put the null check inside the if